docs/topics/collection-elements.md

```diff
@@ -123,6 +123,25 @@ fun main() {
 ```
 {kotlin-runnable="true" kotlin-min-compiler-version="1.3"}
 
+If you need to map the collection before retrieving the element, there is a function [`firstNotNullOf()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/first-not-null-of.html).
+It combines 2 actions:
+- Maps the collection with the selector function
+- Returns the first non-null value in the result
+
+`firstNotNullOf()` throws the `NoSuchElementException` if the resulting collection doesn't have a non-null element. 
+Use the counterpart [`firstNotNullOfOrNull()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/first-not-null-of-or-null.html) 
+to return null in this case.
+
+```kotlin
+fun main() {
+//sampleStart
+    val data = listOf("0", "true", "false")
+    println(data.firstNotNullOf(String::toBooleanStrictOrNull))
```

Comment:
let's make a simple example without callable references and unfamiliar functions. A well-known string operation or smth similar.

Reply:
I propose to show how it could play nicely with `takeIf`

```
val stuff = listOf<Any>(0, true, false)
// converts each element to string and returns the first one that has required length
val longEnough = stuff.firstNotNullOf { it.toString().takeIf { it.length >= 4 } } 
```

Reply:
Thanks! Added this sample.